src/Compilers/VisualBasic/Portable/Binding/ImportAliasesBinder.vb

```diff
@@ -40,6 +41,10 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                            TypeOf containingBinder.ContainingBinder.ContainingBinder Is SourceModuleBinder)))
         End Sub
 
+        Public Function GetImportChainData() As ImmutableArray(Of IAliasSymbol)
+            Return ImmutableArray(Of IAliasSymbol).CastUp(_importedAliases.SelectAsArray(Function(kvp) kvp.Value.Alias))
```

Comment:
>_importedAliases.SelectAsArray [](http://example.com/codeflow?start=58&length=30)

The order doesn't look deterministic

Reply:
Currently, that's intentional.  I've explicitly stated in the docs that order is not guaranteed.  Do you think that's sufficient/  or would you prefer we stated some guarantee here and then coded that in?

Reply:
> Currently, that's intentional.  I've explicitly stated in the docs that order is not guaranteed.  Do you think that's sufficient/  or would you prefer we stated some guarantee here and then coded that in?

In general we are trying to avoid behavior like that in a public API. Even if we do not document any particular order, we are trying to avoid "randomization" for the same scenario, which is likely to happen in this case. I think we don't need to block this PR, but let's bring this issue for discussion at the API review meeting.